### PR TITLE
BSC support

### PR DESCRIPTION
**Pull request by poolpitako**
### What I did

Related issue: #

### Checklist

- [ ] I have confirmed that my PR passes all linting checks
- [ ] I have included test cases
- [ ] I have updated the documentation
- [ ] I have added an entry to the changelog

### CODE REVIEW

File: brownie/network/contract.py [hunk 1]
```diff
@@ -775,8 +775,13 @@ def _check_and_set(self, name: str, obj: Any) -> None:
                 BrownieEnvironmentWarning,
             )
         elif hasattr(self, name):
-            raise AttributeError(f"Namespace collision: '{self._name}.{name}'")
-        setattr(self, name, obj)
+            warnings.warn(
+                f"WARNING: Namespace collision: '{self._name}.{name}'\n"
+                f"You will not be able to call _name() on {self._name}",
```
Comment (iamdefinitelyahuman) [inline brownie/network/contract.py:780]:
```suggestion
                f"The {name} function will not be available when interacting with {self._name}",
```

Comment (iamdefinitelyahuman) [reply brownie/network/contract.py:780]: Generally not a fan of this but I understand why it's needed..  in a v2 need to solve this more elegantly.

File: brownie/data/network-config.yaml [hunk 2]
```diff
@@ -38,6 +38,18 @@ live:
...
         host: https://www.ethercluster.com/kotti
         explorer: https://blockscout.com/etc/kotti/api
+  - name: Binance Smart Chain
+    networks:
+      - name: bsc-test
+        chainid: 97
+        id: bsc-test
+        host: https://data-seed-prebsc-1-s1.binance.org:8545
+        explorer: https://api-testnet.bscscan.com/api
+      - name: bsc-main
```
Comment (iamdefinitelyahuman) [inline brownie/data/network-config.yaml:48]: A more descriptive name than just `bsc-main` would be good.  Same for the testnet.

**Review by iamdefinitelyahuman — CHANGES_REQUESTED** [review]

**Comment by iamdefinitelyahuman** [comment]
Looks like this has caused some tests to fail.

**Review by iamdefinitelyahuman — APPROVED** [review]

**Comment by poolpitako** [comment]
@iamdefinitelyahuman ouch. sorry :(
Thanks for updating the pr!

**Pull request merged**